core/web/health_controller.go

```diff
@@ -93,7 +99,200 @@ func (hc *HealthController) Health(c *gin.Context) {
 			Output: output,
 		})
 	}
+	slices.SortFunc(checks, func(a, b presenters.Check) int {
+		return cmp.Compare(a.Name, b.Name)
+	})
+
+	switch c.NegotiateFormat(gin.MIMEJSON, gin.MIMEHTML, gin.MIMEPlain) {
+	case gin.MIMEJSON:
+		break // default
+
+	case gin.MIMEHTML:
+		if err := newCheckTree(checks).WriteHTMLTo(c.Writer); err != nil {
+			hc.App.GetLogger().Errorw("Failed to write HTML health report", "err", err)
+			c.AbortWithStatus(http.StatusInternalServerError)
+		}
+		return
+
+	case gin.MIMEPlain:
+		slices.SortFunc(checks, func(a, b presenters.Check) int {
+			return cmp.Compare(a.Name, b.Name)
+		})
+
+		if err := writeTextTo(c.Writer, checks); err != nil {
+			hc.App.GetLogger().Errorw("Failed to write plaintext health report", "err", err)
+			c.AbortWithStatus(http.StatusInternalServerError)
+		}
+		return
+	}
 
 	// return a json description of all the checks
-	jsonAPIResponse(c, checks, "checks")
+	jsonAPIResponseWithStatus(c, checks, "checks", status)
+}
+
+func writeTextTo(w io.Writer, checks []presenters.Check) error {
+	slices.SortFunc(checks, func(a, b presenters.Check) int {
+		return cmp.Compare(a.Name, b.Name)
+	})
+	for _, ch := range checks {
+		status := "?"
+		switch ch.Status {
+		case HealthStatusPassing:
+			status = "-"
+		case HealthStatusFailing:
+			status = "!"
+		}
+		if _, err := fmt.Fprintf(w, "%s%s\n", status, ch.Name); err != nil {
+			return err
+		}
+		if ch.Output != "" {
+			if _, err := fmt.Fprintf(newLinePrefixWriter(w, "\t"), "\t%s", ch.Output); err != nil {
+				return err
+			}
+			if _, err := fmt.Fprintln(w); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+type checkNode struct {
+	Name   string // full
+	Status string
+	Output string
+
+	Subs checkTree
+}
+
+type checkTree map[string]checkNode
+
+func newCheckTree(checks []presenters.Check) checkTree {
+	slices.SortFunc(checks, func(a, b presenters.Check) int {
+		return cmp.Compare(a.Name, b.Name)
+	})
```

Comment:
I can drop this now that we always sort...unless that is too costly for the default JSON case?
```suggestion
```